File: PP-Lab6/Lab6/F2.cpp
#include "F2.h"

F2::F2(int N) {
	this->N = N;
}

DWORD F2::run() {
	cout << "T2 started." << endl;
	Data* data = new Data(N);

	vector<vector<int>> MG, MH, MK;

	// Generate Input Values
	Sleep(100);
	cout << "T2 gets a permit." << endl;

	MG = data->FillMatrixWithNumber(2);
	MH = data->FillMatrixWithNumber(2);
	MK = data->FillMatrixWithNumber(2);

	// Calculate The Result
	vector<vector<int>> result = data->Func2(MG, MH, MK);
	Sleep(100);

	// Output
	if (N <= 10) {
		cout << "T2 result:\n";
		data->MatrixOutput(result, "MF");
	}

	cout << "T2 releases the permit." << endl;
	cout << "T2 finished.\n" << endl;

	delete data;

	return 0;
}

File: PP-Lab3/Lab3/F2.cs
using System;
using System.Threading;
using static Lab3.Lab3;

namespace Lab3
{
    public class F2
    {
        // MF = (MG*MH)*TRANS(MK)
        public void Run()
        {
            Console.WriteLine("T2 started.");
            Data data = new Data(N);

            int[ , ] MG, MH, MK;

            // Generate Input Values
            Thread.Sleep(50);
            Console.WriteLine("T2 is waiting for a permit.");

            Lab3.sem.WaitOne();
            Thread.Sleep(100);
            Console.WriteLine("\nT2 gets a permit.\n");

            MG = data.MatrixInput("MG"); 
            MH = data.MatrixInput("MH"); 
            MK = data.MatrixInput("MK");
                
            Console.WriteLine("\nT2 releases the permit.");
            Lab3.sem.Release();
            Console.WriteLine("\nT2 is waiting for a permit.");

            // Calculate The Result
            int[ , ] result = data.Func2(MG, MH, MK);
            Thread.Sleep(100);

            // Output
            Lab3.sem.WaitOne();
            
            Console.WriteLine("T2 gets a permit.\n");
            Console.Write("T2 result:\n");
            data.MatrixOutput(result, "MF");

            Console.WriteLine("T2 releases the permit.");
            Lab3.sem.Release();

            Console.WriteLine("T2 finished.\n");
        }
    }

}
